crates/journal/src/journal.rs

```diff
@@ -66,10 +71,14 @@ pub fn new_journal_entry(workspace: &Workspace, window: &mut Window, cx: &mut Ap
     let journal_dir_clone = journal_dir.clone();
 
     let now = Local::now();
-    let month_dir = journal_dir
-        .join(format!("{:02}", now.year()))
-        .join(format!("{:02}", now.month()));
-    let entry_path = month_dir.join(format!("{:02}.md", now.day()));
+    let entry_path = journal_dir.join(now.format(&settings.date_format).to_string());
+
+    if entry_path.file_name().is_none() {
```

Comment:
This is wrong, directories usually have a file name: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=76bbb3483b8f50dd57a3cc76cda79c52

The wording is also odd: date format to produce a file name?
Why does it even need to do so, and why cannot Zed append this suffix instead?